Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware.swift

```diff
@@ -28,33 +40,80 @@ struct MemberEventMiddleware: EventMiddleware {
                 
                 // We remove the member from the channel
                 channel.members.remove(member)
+                if let membership = channel.membership, membership.user.id == event.user.id {
+                    channel.membership = nil
+                }
```

Comment:
Hm, why we need this part? Won't we receive `notification.removed_from_channel` event?

Reply:
yes, but just to be on the safe side, to safeguard against possible backend bugs

Reply:
Hmm, we might actually need that 🤔 If the current user is watching a channel and it gets removed from the member list, will we get `notification.removed_from_channel` or `member.removed`?

Reply:
the first one

Reply:
If we are getting events it means the current user is connected so, yeah, it will a bug on the backend if the wrong event/no event is fired. I would stick to the approach when `channel.membership` is modified only by `notification.*` events

Reply:
Discussed internally, decided to have the safeguard in place.